pkg/controllers/provisioningv2/rke2/bootstrap/controller.go

```diff
@@ -113,7 +113,15 @@ func (h *handler) getBootstrapSecret(namespace, name string, envVars []corev1.En
 			return nil, err
 		}
 
-		data, err := installer.LinuxInstallScript(context.WithValue(context.Background(), tls.InternalAPI, hasHostPort), base64.URLEncoding.EncodeToString(hash[:]), envVars, "")
+		is := installer.LinuxInstallScript
+		// This is only used by the vmwarevsphere node driver as of Jan 2022
+		// as we add multi-platform support to more node drivers
+		// this should continue to work based on changes in
+		// pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go
+		if os, ok := machine.GetAnnotations()[rke2.MachineOSAnnotation]; ok && os == rke2.WindowsMachineOS {
```

Comment:
```suggestion
		if os := machine.Annotations[rke2.MachineOSAnnotation]; os == rke2.WindowsMachineOS {
```

Reply:
If the annotation doesn't exist, then it won't be the value that you need it to be. It will be empty. Checking both is not necessary.

Reply:
understood, thank you for explaining! Fixing this in my next push